docs/standard/analyzers/api-analyzer.md

```diff
@@ -0,0 +1,62 @@
+author: oliag
+ms.author: mairaw
+ms.date: 12/11/2017
+ms.topic: article
+ms.prod: .net
+ms.technology: dotnet-standard
+---
+
+# .NET API Analyzer
+
+The .NET API Analyzer is a Roslyn analyzer that discovers potential compatibility risks for APIs with different platforms and detecting calls to deprecated APIs. It can be useful for any developers of the .NET family at any stages of development.
+
+API Analyzer comes as [NuGet package](https://www.nuget.org/packages/Microsoft.DotNet.Analyzers.Compatibility/) and after referencing it in a project, it automatically starts monitoring the code and squiggles problematic API usage. You can get suggestions on possible fixes by clicking on the light bulb, which also includes the ability to suppress the warnings.
+
+## Discovering deprecated APIs
+When a deprecated API (for example `WebClient`) is used in a code, API Analyzer squiggles it out in green and shows a light bulb on the left
+
+![](media/api-analyzer/green-squiggle.jpg) !
+
+The Error List window contains warnings with a unique ID per deprecated API (in our example `DE004`). By clicking on it, you go to a webpage with detailed information about why the API was deprecated and how alternative APIs should be used.
+
+![](media/api-analyzer/warnings.jpg)
+
+Any warnings can be suppressed by right-clicking on the highlighted member and selecting **Quick Actions and Refactorings**. Here you get two suppression options: locally (in source) or globally (in a suppression file). We encourage developers to use global suppression to ensure consistency of API usage across their project.
+
+## Discovering cross-platform issues
+Similar to deprecated APIs the analyzer squiggles out all APIs that are not cross-platform supported (For example `Console.WindowWidth` works on Windows but not on Linux and macOS). The diagnostic ID is shown in Error List window. You can suppress warning by right click and choosing "Quick Actions and Refactorings". Unlike deprecation cases where you have two options: either keep using deprecated member and suppress warnings or not use it at all, here if you are developing your code only for certain platform you can suppress all warnings for all other  platforms you don't plan to run your code on. To do so, you just need to edit your project file and add a property `PlatformCompatIgnore` that lists all platforms to be ignored:
+```
+<PropertyGroup>
+    <PlatformCompatIgnore>Linux;MacOSX</PlatformCompatIgnore>
+</PropertyGroup>
+```
+If your code targets multiple platforms and you want to take advantage of an API not supported on some of them, you can guard that part of the code with an `if`-statement:
+
+```CSharp
+if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+{
+     var w = Console.WindowWidth;
+     // More code
+}
+```
+You can also conditionally compile per target framework/operating system, but you currently need to do that manually.
```

Comment:
Can you add a bit more detail about what you mean by doing that manually?

Reply:
I agree with @rpetrusha that more details here would be good.